docs/how-to-build-a-plugin.md

```diff
@@ -0,0 +1,475 @@
+    this.logger.debug("Reset collection");
+    this._myCollection = [this.config.myValue];
+  }
+
+  _addBindings() {
+    this.eventManager.listen(this.player, this.player.Event.SEEKED, () => {
+      this.logger.debug(this.player.currentTime + " Added to my collection");
+      this._myCollection.push(this.player.currentTime);
+      this.dispatchEvent("collectionUpdate", {collection: this._myCollection});
+    });
+  }
+}
+```
+
+## Step 5 - Register Your Plugin
+
+Use the factory `registerPlugin` method to register your plugin in the player framework:
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>MyPlugin</title>
+    <script src="PATH/TO/PLAYER/LIB/FILENAME.js"></script>
+    <script src="PATH/TO/FILE/my-plugin.js"></script>
+</head>
+<body>
+<script>
+	KalturaPlayer.core.registerPlugin("myPlugin", MyPlugin);
+</script>
+</body>
+</html>
+```
+
+## Step 5 - Configure Your Plugin
+
+All that's left now is to verify that the player activates the plugin during runtime.
+
+1 . Include the new plugin name and plugin configuration in the player configuration. For example:
+```js
+var config = {
+    player: {
+        plugins: {
+            myPlugin: {
+                myValue: 10
+            }
+        }
+    }
+};
+```
+
+2 . Let's expand our test page and add:
+
+a . A placeholder div to the player:
+```html
+<div id="player-div" style="width: 360px; height: 640px;"></div>
+```
+b . A basic player configuration:
+```js
+var config = {
+    logLevel: "DEBUG",
+    targetId: "player-div",
+    provider: {
+        partnerId: "YOUR_PARTNER_ID"
+    },
+    player: {
+        plugins: {
+            myPlugin: {
+                myValue: 10
+            }
+        }
+    }
+};
+```
+c . Setup code.
+```js
+var mediaInfo = {entryId: "YOUR_ENTRY_ID"};
+var player = KalturaPlayer.setup(config);
+player.loadMedia(mediaInfo).then(function() {
+  player.play();
+});
+```
+
+3 . Connect everything together and you'll get:
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>MyPlugin</title>
+    <script src="PATH/TO/PLAYER/LIB/FILENAME.js"></script>
+    <script src="PATH/TO/FILE/my-plugin.js"></script>
+</head>
+<body>
+<script>
+KalturaPlayer.core.registerPlugin("myPlugin", MyPlugin);
+
+var config = {
+    logLevel: "DEBUG",
+    targetId: "player-div",
+    provider: {
+        partnerId: "YOUR_PARTNER_ID"
+    },
+    player: {
+        plugins: {
+            myPlugin: {
+                myValue: 10
+            }
+        }
+    }
+};
+
+var mediaInfo = {entryId: "YOUR_ENTRY_ID"};
+var player = KalturaPlayer.setup(config);
+player.loadMedia(mediaInfo).then(function() {
+  player.play();
+});
+</script>
+</body>
+</html>
+```
+
+**That's it! the player starts to play and your plugin is activated.**
+
+# Writing an Advanced Plugin
+
+An advanced plugin is one that is built as a self project and handles a version, tests, documentation, dev scripts and so on.
```

Comment:
@dan-ziv - what do you mean by "self project"?
@OrenMe - I know we want to keep the docs short, but in this case, I think we have to live with a longer document or split this into a series of pages, which may be a bit confusing. WDYT?

Reply:
maybe `standalone github repo` is more suitable.
this is one of the more technical oriented docs so we can keep it as is